app/src/test/java/org/fossasia/openevent/general/AppLinkUtilsTest.kt

```diff
@@ -0,0 +1,54 @@
+package org.fossasia.openevent.general
+
+import org.fossasia.openevent.general.utils.EVENT_IDENTIFIER
+import org.fossasia.openevent.general.utils.RESET_PASSWORD_TOKEN
+import org.fossasia.openevent.general.utils.AppLinkUtils
+import org.fossasia.openevent.general.utils.VERIFICATION_TOKEN
+import org.junit.Test
+import org.junit.Assert.assertEquals
+
+private const val EVENT = "event"
+private const val RESET_PASSWORD = "resetPassword"
+private const val VERIFY_EMAIL = "verifyEmail"
+
+class AppLinkUtilsTest {
+
+    private fun getAppLink(type: String): String {
+        return when (type) {
+            EVENT -> "https://eventyay.com/e/5f6d3feb"
+            RESET_PASSWORD -> "https://eventyay.com/reset-password?token=822980340478781748445098077144"
+            VERIFY_EMAIL -> "https://eventyay.com/verify?token=WyJsaXZlLmhhcnNoaXRAaG"
+            else -> ""
+        }
+    }
+
+    @Test
+    fun `should get event link`() {
+        val uri = getAppLink(EVENT)
+        assertEquals(R.id.eventDetailsFragment, AppLinkUtils.getData(uri)?.destinationId)
+        assertEquals(EVENT_IDENTIFIER, AppLinkUtils.getData(uri)?.argumentKey)
+        assertEquals("""
+            5f6d3feb
+        """.trimIndent(), AppLinkUtils.getData(uri)?.argumentValue)
```

Comment:
`assertEquals(AppLinkData(a, b, c), AppLinkUtils.getData(uri))`

convert to a single call